File: JavaScript/3d_ex/main.js
function err(exc) {
    throw new Error(exc);
}

function init() {
    const val = parseInt(prompt());
    if (isNaN(val)) {
        err("Invalid input");
    }
    return val;
}

function countingDup(n) {
    let a = init();
    let b = a;
    let count = 0;

    for (let i = 1; i < n; i++) {
        a = init();
        if (a === b) {
            count += 1;
        }
        b = a;
    }
    return count;
}
try {
    const n = init();
    if (n <= 0) {
        err("Invalid input");
    }

    console.log(countingDup(n));
} catch (e) {
    console.error(e.message);
}

File: ruby/3d_ex/main.rb
def err(exc)
  raise StandardError, exc
end

def init
  gets.to_i
end

def counting_dup(n)
  a = b = init
  count = 0

  (n - 1).times do
    a = init
    count += 1 if a == b
    b = a
  end
  count
end

n = init
err("Invalid input") if n <= 0

puts counting_dup(n)
